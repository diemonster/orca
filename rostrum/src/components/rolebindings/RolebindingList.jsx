import React from 'react';
import { connect } from 'react-redux';

import { rolebindingList } from '../../actions/rolebindings';

import RolebindingListItem from './RolebindingListItem';

class RolebindingList extends React.Component {
  componentDidMount() {
    const { rolebindingList, namespace } = this.props;
    rolebindingList(namespace);
  }

  render() {
    // namespace is passed in already from the parent function
    // rolebindingObjects is populated through rolebindingList()
    const { namespace, rolebindingObjects } = this.props;

    var namespacedRolebindings = null;
    if (rolebindingObjects) {
      namespacedRolebindings = rolebindingObjects[namespace];
    }

    var rolebindingListItems = null;
    if (namespacedRolebindings) {
      rolebindingListItems = namespacedRolebindings.map((rolebinding, idx) =>
        <RolebindingListItem key={idx} rolebinding={rolebinding.metadata.name} />
      );
    }

    return (
      <div className='rolebinding-container'>
        <h5>Role Bindings:</h5>
        <ul className='rolebinding-list'>
          {rolebindingListItems}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    rolebindingObjects: state.rolebinding.rolebindingObjects
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    rolebindingList: (namespace) => dispatch(rolebindingList(namespace))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(RolebindingList);
