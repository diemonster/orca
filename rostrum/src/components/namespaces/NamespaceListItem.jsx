import React from 'react';
import PropTypes from 'prop-types';

import RolebindingList from '../rolebindings/RolebindingList';

function NamespaceListItem(props) {
  const { namespace, phase } = props;

  let cls = null;
  switch (phase) {
    case 'Terminating':
      cls = 'phase-terminating';
      break;
    default:
      cls = 'phase-active';
  }

  return (
    <li className={`namespace-list-item ${cls}`}>
      <h4>{namespace}</h4>
      <RolebindingList namespace={namespace} />
    </li>
  );
}

NamespaceListItem.propTypes = {
  namespace: PropTypes.string.isRequired,
  phase: PropTypes.string.isRequired,
};

export default NamespaceListItem;
