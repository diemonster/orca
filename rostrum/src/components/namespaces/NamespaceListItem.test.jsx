import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NamespaceListItem from './NamespaceListItem';
import RolebindingList from '../rolebindings/RolebindingList';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('NamespaceListItem', () => {
    it('should render itself and its subcomponents', () => {
      const props = {
        namespace: 'some-namespace',
        phase: 'Active',
      };

      const namespaceListItem = shallow(<NamespaceListItem {...props} />);

      expect(namespaceListItem).toMatchSnapshot();
      expect(namespaceListItem.find('li').hasClass('namespace-list-item')).toBe(true);
      expect(namespaceListItem.find('li').hasClass('phase-active')).toBe(true);
      expect(namespaceListItem.find('h4').text()).toEqual('some-namespace');

      const rolebindingListProps = namespaceListItem.find(RolebindingList).props();
      expect(rolebindingListProps.namespace).toEqual('some-namespace');
    });

    it('should set "phase-terminating" class if passed "Terminating"', () => {
      const props = {
        namespace: 'some-namespace',
        phase: 'Terminating',
      };

      const namespaceListItem = shallow(<NamespaceListItem {...props} />);

      expect(namespaceListItem.find('li').hasClass('phase-terminating')).toBe(true);
    });
  });
});
