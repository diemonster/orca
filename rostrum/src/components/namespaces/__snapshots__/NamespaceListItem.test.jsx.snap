// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NamespaceListItem should render itself with a terminating namespace 1`] = `
<li
  className="namespace-list-item phase-terminating"
>
  <h4>
    some-namespace
  </h4>
  <Connect(RolebindingList)
    namespace="some-namespace"
  />
</li>
`;

exports[`components NamespaceListItem should render itself with an active namespace 1`] = `
<li
  className="namespace-list-item phase-active"
>
  <h4>
    some-namespace
  </h4>
  <Connect(RolebindingList)
    namespace="some-namespace"
  />
</li>
`;
